### PR TITLE
Add MinMaxScaler component for normalization

### LINKED ISSUE #3355: Add a normalization component
Part of the work for supporting unsupervised learning in evalml, we need a normalization component to ensure features on different scales don't negatively affect the performance of models.

### PR DESCRIPTION
**Pull request by eccabay**
Closes #3355

### CODE REVIEW

**Review by chukarsten — APPROVED** [review]
LGTM Becca!  I think just need to add those asserts showing that the dataset values already falls outside the MinMaxScaler's default range.

File: evalml/tests/component_tests/test_minmax_scaler.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        assert max(X_t[col]) <= 1.0
+
+    scaler = MinMaxScaler(feature_range=(-1, 3))
+    X_t = scaler.fit_transform(X, y)
+    for col in X_t.columns:
+        assert -1 <= min(X_t[col]) < 0
+        assert 1 < max(X_t[col]) <= 3.1
+
+
+def test_minmax_scaler_numeric_and_categorical():
```
Comment (jeremyliweishih) [inline evalml/tests/component_tests/test_minmax_scaler.py:25]: might be a good idea to check that woodwork inference isn't lost in this test.

**Review by jeremyliweishih — APPROVED** [review]
Left one testing comment, looks great!

File: evalml/tests/component_tests/test_minmax_scaler.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    X_t = scaler.transform(X)
+    for col in X_t.columns:
+        assert min(X_t[col]) >= 0.0
+        assert max(X_t[col]) <= 1.0
+
+    scaler = MinMaxScaler(feature_range=(-1, 3))
+    X_t = scaler.fit_transform(X, y)
+    for col in X_t.columns:
+        assert -1 <= min(X_t[col]) < 0
+        assert 1 < max(X_t[col]) <= 3.1
```
Comment (bchen1116) [inline evalml/tests/component_tests/test_minmax_scaler.py:22]: Why is the range here 3.1 instead of 3?

File: evalml/tests/component_tests/test_minmax_scaler.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            "col_1": ["a", "a", "b", "b", "a"],
+            "col_2": ["a", "b", "a", "c", "b"],
+            "col_3": [False, True, True, False, True],
+        }
+    )
+    scaler = MinMaxScaler()
+    scaler.fit(X)
+    X_t = scaler.transform(X)
+
+    pd.testing.assert_frame_equal(X, X_t, check_dtype=False)
```
Comment (bchen1116) [inline evalml/tests/component_tests/test_minmax_scaler.py:56]: just a nit, but can we do `assert scaler._is_fitted` to ensurre that the scaler is fitted?

**Review by bchen1116 — APPROVED** [review]
Left a question on a test and a suggestion, but LGTM! Can you add this component, and the other Unsupervised components if applicable, to the `api_reference.rst` file?

File: evalml/tests/component_tests/test_minmax_scaler.py (hunk 2 above)
Comment (eccabay) [reply evalml/tests/component_tests/test_minmax_scaler.py:22]: There's a small rounding error, so the max value in this case is actually 3.00004. Close enough to not matter, but far enough to break the test if left at 3.

Comment (bchen1116) [reply evalml/tests/component_tests/test_minmax_scaler.py:22]: Wow, that's interesting that the rounding error would pass sklearn's internal tests. 
This seems fine to me then!
